src/dotnet/commands/dotnet-restore3/Restore3CommandApp.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using Microsoft.DotNet.Cli;
+using Microsoft.DotNet.Cli.CommandLine;
+
+namespace Microsoft.DotNet.Tools.Restore3
+{
+    public class Restore3CommandApp
+    {
+        public static int Execute(string[] args)
+        {
+            CommandLineApplication cmd = new CommandLineApplication
+            {
+                Name = "restore3",
+                FullName = "restore3",
+                Description = "restore for msbuild"
+            };
+
+            cmd.HelpOption("-h|--help");
+
+            var argRoot = cmd.Argument(
+                    "[root]",
+                    "Optional path to a project file or MSBuild arguments.",
+                    multipleValues: true);
+
+            var sourceOption = cmd.Option(
```

Comment:
``-s`` just overrides, I think we need to look at fallback source, that scenario can be useful although it is rare, and at minimum it needs a new name to help with #2477. I'm not willing to add it back under the fallback name here since this is a new command and it should stay as clean as possible.